src/app/archive/upgrade-to-mesa.sql

```diff
@@ -160,7 +161,56 @@ SELECT add_zkapp_states_nullable_element(29);
 SELECT add_zkapp_states_nullable_element(30);
 SELECT add_zkapp_states_nullable_element(31);
 
--- 3. Update schema_history
+-- 3. `zkapp_states`: Add columns element8..element31
+
+CREATE OR REPLACE FUNCTION add_zkapp_states_element(p_element_num INT)
+RETURNS VOID AS $$
+DECLARE
+    col_name TEXT := 'element' || p_element_num;
+BEGIN
+
+    RAISE DEBUG 'Adding column % for zkapp_states', col_name;
+
+    EXECUTE format(
+        'ALTER TABLE zkapp_states ADD COLUMN IF NOT EXISTS %I INT DEFAULT 0 NOT NULL REFERENCES zkapp_field(id)',
```

Comment:
Can you reason why 0 not null?

Reply:
MIP explicitly specify we should use 0 as polyfill.